### ISSUE TITLE
IBM AIX

### ISSUE THREAD

**Issue by ArtiomWin** [86Box/86Box#322]
I have problems with install of AIX 1.3: when I put Install floppy, it gives 911-455 KERNEL TRAP.
Config is included: [AIX PS2.zip](https://github.com/86Box/86Box/files/2425891/AIX.PS2.zip)

**Comment by OBattler** [comment]
I'd need the NVR folder as well.

**Comment by ArtiomWin** [comment]
OK, here you are.
[nvr.zip](https://github.com/86Box/86Box/files/2427291/nvr.zip)

**Comment by OBattler** [comment]
This looks like a bug with the IBM MCA ESDI controller.

**Comment by ArtiomWin** [comment]
I've tried to install AIX again on recent build, this time on SCSI. Result is "Error in obtating information on hard disk drives available" (this issue was on earlier builds also). NVR and configs included.
![20190921_002359](https://user-images.githubusercontent.com/31134876/65359767-31819080-dc06-11e9-9bef-c06d572f3070.png)
[nvr+cfg.zip](https://github.com/86Box/86Box/files/3637810/nvr%2Bcfg.zip)

**Comment by TC1995** [comment]
I think AIX doesn't support the current emulated SCSI cards, iirc it only supports the IBM SCSI card, which is not emulated and also poorly documented.

**Comment by ArtiomWin** [comment]
Well, documentation for IBM SCSI adapters is harder to find, but at least that's not 8514 adapter. Here's technical reference for IBM PS/2 SCSI Adapter with Cache: http://www.mcamafia.de/pdf/ibm_scsiwc_trm.pdf

**Comment by waltje** [comment]
Correct.


     On Saturday, September 21, 2019, 6:46:14 AM EDT, TC1995 <notifications@github.com> wrote:  
 
 
I think AIX doesn't support the current emulated SCSI cards, iirc it only supports the IBM SCSI card, which is not emulated and also poorly documented.

**Comment by OBattler** [comment]
The IBM PS/2 SCSI Adapter is going to be implemented in v2.10 based ont he technical reference manual.

**Comment by ArtiomWin** [comment]
Well, I'm glad I made a contribution, but I'm going to return to question of ESDI: is it going to be fixed?

**Comment by OBattler** [comment]
The IBM ESDI controller is going to be fixed at some point in the future.

**Comment by qeeg** [comment]
I'm still getting the kernel trap bug with the IBM MCA SCSI controller selected...

**Comment by Dizzy611** [comment]
Attempted to reproduce this bug but could not find a version of IBM AIX 1.3 that was bootable, with the one I found on WinWorld having "Non-system disk or disk error" for the ESDI boot disks and install disk.

**Comment by ArtiomWin** [comment]
> Attempted to reproduce this bug but could not find a version of IBM AIX 1.3 that was bootable, with the one I found on WinWorld having "Non-system disk or disk error" for the ESDI boot disks and install disk.
https://mega.nz/file/FA9GRQLJ#Mg5p8O032DB7legV2rShvgjMAS0qkVyflDDLHWl6DNc

**Comment by Dizzy611** [comment]
Thank you for the link! These disks are bootable this time. I _think_ I'm getting the same issue with ESDI but I'm not sure to be honest, because it gets partway into the boot process and then immediately reboots before I can read what the error onscreen is. It at least gets to this point first:

![image](https://user-images.githubusercontent.com/301699/164947869-e4f70829-866e-4e88-9d8a-c8b7fb313943.png)

**Comment by ArtiomWin** [comment]
> Thank you for the link! These disks are bootable this time. I _think_ I'm getting the same issue with ESDI but I'm not sure to be honest, because it gets partway into the boot process and then immediately reboots before I can read what the error onscreen is.

I had to use screen recorder, but here's that screen
![Error](https://user-images.githubusercontent.com/31134876/164965667-177c7891-cbc9-4f33-940a-0660fc2b7b2e.png)
:

**Comment by rakslice** [comment]
When using IDE configurations with the ISA support from PTF0024 which you have in your disk set there, what is the difference between boards that have a similar kernel trap at roughly the same point in the initial boot (many boards), and ones that work fine (e.g. socket 2 SiS PS/ValuePoint, socket 7 single voltage hp vectra vl 5 series 4, slot 1 asus p3b-f, slot 2 supermicro s2dge)? Is it some kind of firmware/initialization difference? Anyway, I wonder if this is related to the kernel trap on these ESDI configs.

**Comment by rakslice** [comment]
Note that AIX 1.3 with the PTF 0024 update supports some kind of Adaptec ISA SCSI support (see the one line note to this effect in [the PTF 0024 patch notes](http://ohlandl.ipv7.net/AIX_1-3/AIX_PTF0024_U436295.html)). And I can confirm this because it works with the BusLogic driver's AHA-154x compatibility mode in VirtualBox, but only if you put it in the specific "AHA-1540B" mode, which enables a small set of differences (basically a slightly different board ID and disabling a bunch of commands supported by the BT-958D that the AHA-1540 doesn't have. See the `DEV_AHA_1540B` cases in https://www.virtualbox.org/svn/vbox/trunk/src/VBox/Devices/Storage/DevBusLogic.cpp

**Comment by rakslice** [comment]
No, i don't think that's it, there must be some more preliminary probe that isn't working on 86box, as enabling the `ENABLE_AHA154X_LOG` output it's not even getting hit at all as the AIX driver passes the point when it would detect the card.

**Comment by OBattler** [comment]
What you need to enable is ENABLE_X54X_LOG instead.
